src/io/textfile_decoder.cc

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "singa/io/decoder.h"
+#include <string>
+
+namespace singa {
+
+std::vector<Tensor> TextDecoder::Decode(std::string value) {
+  std::vector<Tensor> output;
+  const char* str = value.c_str();
+  const float* d = reinterpret_cast<const float*>(str);
```

Comment:
You cannot cast a string to float array...
The string is like "1.2, 2.3, 4.0, ...".
you can use the tokenizer https://github.com/apache/incubator-singa/blob/dev/include/singa/utils/string.h#L77
to extract the float str one by one and convert them into float values.